api/src/main/java/io/kafbat/ui/service/rbac/extractor/RbacLdapAuthoritiesExtractor.java

```diff
@@ -36,7 +38,7 @@ protected Set<GrantedAuthority> getAdditionalRoles(DirContextOperations user, St
         .stream()
         .filter(r -> r.getSubjects()
             .stream()
-            .filter(subject -> subject.getProvider().equals(Provider.LDAP))
+            .filter(subject -> SUPPORTED_PROVIDERS.contains(subject.getProvider()))
```

Comment:
Do we really need this if RbacLdapAuthoritiesExtractor is not used when working with Active Directory?

Reply:
Aaah, this makes a lot more sense now!

The reason you're encountering the issue with roles being assigned only if they match the names of AD groups is that we use the `DefaultActiveDirectoryAuthoritiesPopulator` which maps AD groups one-to-one without the matching logic we have in `RbacLdapAuthoritiesExtractor`. After all, we need some custom code, adjusting `RbacLdapAuthoritiesExtractor` to support extracting original groups from LDAP and AD.

Reply:
> After all, we need some custom code, adjusting RbacLdapAuthoritiesExtractor to support extracting original groups from LDAP and AD.

Do you need help with this?